casper/src/test/scala/coop/rchain/casper/helper/TaskCasper.scala

```diff
@@ -0,0 +1,113 @@
+    implicit val errorHandlerEff  = ApplicativeError_.applicativeError[Task, CommError](appErrTask)
+    implicit val nodeDiscoveryEff = new NodeDiscoveryStub[Task]()
+    implicit val transportLayerEff =
+      new TransportLayerTestImpl[Task](HashSetCasperTestNode.peerNode("taskNode", 40400),
+                                       Map.empty[PeerNode, mutable.Queue[Protocol]])
+    implicit val metricEff = new Metrics.MetricsNOP[Task]
+    implicit val blockStoreEff =
+      LMDBBlockStore.create[Task](LMDBBlockStore.Config(blockStoreDir, 1024L * 1024))(syncInstance,
+                                                                                      metricEff)
+    implicit val turanOracleEffect = SafetyOracle.turanOracle[Task]
+
+    val activeRuntime  = Runtime.create(runtimeDir, 1024L * 1024)
+    val runtimeManager = RuntimeManager.fromRuntime(activeRuntime)
+    val validatorId    = ValidatorIdentity(Ed25519.toPublic(sk), sk, "ed25519")
+
+    val casperTask = for {
+      _        <- blockStoreEff.put(genesis.blockHash, genesis)
+      blockMap <- blockStoreEff.asMap()
+    } yield
+      MultiParentCasper.hashSetCasper[Task](runtimeManager, Some(validatorId), genesis, blockMap)
+
+    def cleanUp(): Unit = {
+      activeRuntime.close()
+      blockStoreEff.close()
+      runtimeDir.recursivelyDelete()
+      blockStoreDir.recursivelyDelete()
+    }
+
+    (casperTask, cleanUp _)
+  }
+
+  private val appErrTask = new ApplicativeError[Task, CommError] {
+    def ap[A, B](ff: Task[A => B])(fa: Task[A]): Task[B] =
+      ApplicativeError[Task, Throwable].ap[A, B](ff)(fa)
+    def pure[A](x: A): Task[A] = ApplicativeError[Task, Throwable].pure[A](x)
+    def raiseError[A](e: CommError): Task[A] =
+      ApplicativeError[Task, Throwable].raiseError(new Exception(e.toString))
+
+    def handleErrorWith[A](fa: Task[A])(f: (CommError) => Task[A]): Task[A] = fa
+  }
+
+  private val syncInstance =
+    new Sync[Task] {
+      def suspend[A](thunk: => Task[A]): Task[A] = Task.defer(thunk)
+
+      def flatMap[A, B](fa: Task[A])(f: A => Task[B]): Task[B] =
+        implicitly[FlatMap[Task]].flatMap(fa)(f)
+
+      def tailRecM[A, B](a: A)(f: A => Task[Either[A, B]]): Task[B] =
+        implicitly[FlatMap[Task]].tailRecM(a)(f)
+
+      def raiseError[A](e: Throwable): Task[A] = Task.raiseError(e)
+
+      def pure[A](x: A): Task[A] = implicitly[Applicative[Task]].pure(x)
+
+      def bracketCase[A, B](acquire: Task[A])(use: A => Task[B])(
+          release: (A, ExitCase[Throwable]) => Task[Unit]): Task[B] =
+        acquire flatMap { state =>
+          try {
+            use(state)
+          } catch {
+            case t: Throwable => raiseError(t)
+          } finally {
+            release(state, ExitCase.Completed)
+          }
+        }
+
+      def handleErrorWith[A](fa: Task[A])(f: Throwable => Task[A]): Task[A] =
+        ApplicativeError[Task, Throwable].handleErrorWith(fa)(f)
+    }
+
+}
```

Comment:
OK, I see this is only used in the test but we should strive for the code quality anyway. This means DRY principle apply here as well. I think that `import monix.eval.Task.catsAsync` will suffice, this should provide all the instances you need.